packages/xo-server/src/xo-mixins/migrate-vm.mjs

```diff
@@ -106,4 +125,153 @@ export default class MigrateVm {
+        name_description: 'from esxi',
+        name_label,
+        VCPUs_at_startup: numCpu,
+        VCPUs_max: numCpu,
+      })
+    )
+    await Promise.all([
+      asyncMapSettled(['start', 'start_on'], op => vm.update_blocked_operations(op, 'OVA import in progress...')),
+      vm.set_name_label(`[Importing...] ${name_label}`),
+    ])
+
+    const vifDevices = await xapi.call('VM.get_allowed_VIF_devices', vm.$ref)
+
+    await Promise.all(
+      networks.map((network, i) =>
+        xapi.VIF_create({
+          device: vifDevices[i],
+          network: xapi.getObject(networkId).$ref,
+          VM: vm.$ref,
+        })
+      )
+    )
+
+    const vhds = await Promise.all(
+      Object.keys(chainsByNodes).map(async (node,userdevice) =>{
+        const chainByNode = chainsByNodes[node]
+        const vdi = await xapi._getOrWaitObject(
+          await xapi.VDI_create({
+            name_description: 'fromESXI' + chainByNode[0].descriptionLabel,
+            name_label: '[ESXI]' + chainByNode[0].nameLabel,
+            SR: sr.$ref,
+            virtual_size: chainByNode[0].capacity,
+          })
+        )
+        console.log('vdi created')
+
+        await xapi.VBD_create({
+          userdevice: String(userdevice),
+          VDI: vdi.$ref,
+          VM: vm.$ref,
+        })
+        console.log('vbd created')
+        let parentVhd, vhd
+        const nbColdDisks  = isRunning ? chainByNode.length -1 : chainByNode.length
+        console.log('will transfer', nbColdDisks, isRunning)
+        for (let diskIndex = 0; diskIndex < nbColdDisks; diskIndex ++ ) {
+          // the first one  is a RAW disk ( full )
+          const disk= chainByNode[diskIndex]
+          console.log('will import ', { disk })
+          const {fileName, path, datastore,isFull} = disk
+
+          if(isFull){
+            console.log('full disk ')
+            vhd = await VhdEsxiRaw.open(esxi, datastore, path + '/' + fileName, {thin})
+            await vhd.readBlockAllocationTable()
+          } else {
+            console.log('delta disk ')
+            vhd = await VhdCowd.open(esxi, datastore, path + '/' + fileName, parentVhd)
+            await vhd.readBlockAllocationTable()
+          }
+          parentVhd = vhd
+        }
+        console.log('will import synthetic ')
+        await vdi.$importContent(vhd.stream(), { format:VDI_FORMAT_VHD })
+        return {vdi ,vhd}
+      }
+      )
+    )
+    console.log('cold transfer done')
+    if(isRunning){
+      await esxi.powerOff(vmId)
+
+      console.log('vm stopped')
+      await Promise.all(
+        Object.keys(chainsByNodes).map(async (node, userdevice)=>{
+          const chainByNode = chainsByNodes[node]
+          const disk = chainByNode[chainByNode.length -1]
+          const {fileName, path, datastore } = disk
+          const {vdi, vhd:parentVhd} = vhds[userdevice]
+          const vhd = await VhdCowd.open(esxi, datastore, path + '/' + fileName, parentVhd, {chain:false})
```

Comment:
IMHO it would be nice to find a better name that `chain`, maybe `lookMissingBlockInParent`?